### PR TITLE
feat(condo): DOMA-5891 create resetOrganization service

### PR DESCRIPTION
**Pull request by VKislov**

### CODE REVIEW

File: apps/condo/domains/organization/schema/ResetOrganizationService.test.js [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+const { createTestB2BAppContext, B2BAppContext, createTestB2BApp } = require('@condo/domains/miniapp/utils/testSchema')
+const { DELETED_ORGANIZATION_NAME } = require('@condo/domains/organization/constants/common')
+const { resetOrganizationByTestClient, createTestOrganization, Organization, createTestOrganizationEmployee, createTestOrganizationEmployeeRole, createTestOrganizationLink, OrganizationLink } = require('@condo/domains/organization/utils/testSchema')
+const { OrganizationEmployee } = require('@condo/domains/organization/utils/testSchema')
+const { createTestProperty, Property } = require('@condo/domains/property/utils/testSchema')
+const { makeClientWithSupportUser } = require('@condo/domains/user/utils/testSchema')
+const { makeClientWithStaffUser } = require('@condo/domains/user/utils/testSchema')
+
+
+describe('ResetOrganizationService', async () => {
```
Comment (SavelevMatthew) [inline apps/condo/domains/organization/schema/ResetOrganizationService.test.js:20]: 😡 😡 😡 😡 😡

File: apps/condo/domains/organization/schema/ResetOrganizationService.js [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    ],
+    
+    mutations: [
+        {
+            access: access.canResetOrganization,
+            schema: 'resetOrganization(data: ResetOrganizationInput!): ResetOrganizationOutput',
+            resolver: async (parent, args, context, info, extra = {}) => {
+                const { data: { dv, sender, organizationId } } = args
+
+                if (dv !== 1) {
```
Comment (pahaz) [inline apps/condo/domains/organization/schema/ResetOrganizationService.js:65]: it's better to use `const { checkDvAndSender } = require('@open-condo/keystone/plugins/dvAndSender')`. Look at example from `ConfirmPhoneActionService`

File: apps/condo/domains/organization/schema/ResetOrganizationService.js [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+
+                if (!organizationId) throw new Error('resetOrganization(): no organizationId')
+
+                const organizationEntity = await getById('Organization', organizationId)
+                if (!organizationEntity) {
+                    throw new GQLError(errors.ORGANIZATION_NOT_FOUND, context)
+                }
+
+                const properties = await loadListByChunks({
+                    context: context,
```
Comment (pahaz) [inline apps/condo/domains/organization/schema/ResetOrganizationService.js:81]:
```suggestion
                    context,
```

File: apps/condo/domains/organization/schema/ResetOrganizationService.js [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+
+                const acquiringIntegrationCtxs = await AcquiringIntegrationContext.getAll(context, {
+                    deletedAt: null,
+                    organization: { id: organizationId },
+                })
+                for (let acquiringIntegrationCtx of acquiringIntegrationCtxs) {
+                    await AcquiringIntegrationContext.softDelete(context, acquiringIntegrationCtx.id, DV_SENDER)
+                }
+
+                const b2BAppCtxs = await B2BAppContext.getAll(context, {
```
Comment (pahaz) [inline apps/condo/domains/organization/schema/ResetOrganizationService.js:135]: Probably you also need to delete BankingContext

File: apps/condo/domains/organization/schema/ResetOrganizationService.test.js (hunk 1 above)
Comment (pahaz) [inline apps/condo/domains/organization/schema/ResetOrganizationService.test.js:20]:
```suggestion
describe('ResetOrganizationService', () => {
```

File: apps/condo/domains/organization/schema/ResetOrganizationService.js (hunk 2 above)
Comment (pahaz) [reply apps/condo/domains/organization/schema/ResetOrganizationService.js:65]: Probably it's better to fix the same in `resetUser`

**Review by pahaz — APPROVED** [review]

**Review by sitozzz — APPROVED** [review]

File: apps/condo/domains/organization/schema/ResetOrganizationService.js [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+const access = require('@condo/domains/organization/access/ResetOrganizationService')
+const { DELETED_ORGANIZATION_NAME } = require('@condo/domains/organization/constants/common')
+const { Organization, OrganizationLink, OrganizationEmployee } = require('@condo/domains/organization/utils/serverSchema')
+const { Property } = require('@condo/domains/property/utils/serverSchema')
+
+/**
+ * List of possible errors, that this custom schema can throw
+ * They will be rendered in documentation section in GraphiQL for this custom schema
+ */
+const errors = {
```
Comment (Alllex202) [inline apps/condo/domains/organization/schema/ResetOrganizationService.js:24]:
```suggestion
const ERRORS = {
```

File: apps/condo/domains/organization/schema/ResetOrganizationService.js [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+                // banking domain
+                const bankIntegrationOrganizationContexts = await BankIntegrationOrganizationContext.getAll(context, {
+                    deletedAt: null,
+                    organization: { id: organizationId },
+                })
+                for (let bankIntegrationOrganizationContext of bankIntegrationOrganizationContexts) {
+                    await BankIntegrationOrganizationContext.softDelete(context, bankIntegrationOrganizationContext.id, DV_SENDER)
+                }
+
+                const BankIntegrationAccountCtxs = await loadListByChunks({
```
Comment (Alllex202) [inline apps/condo/domains/organization/schema/ResetOrganizationService.js:149]:
```suggestion
                const bankIntegrationAccountCtxs = await loadListByChunks({
```

**Review by Alllex202 — APPROVED** [review]

File: apps/condo/domains/organization/schema/ResetOrganizationService.js [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+                    limit: 100000,
+                    where: {
+                        deletedAt: null,
+                        organization: { id: organizationId },
+                    },
+                })
+                for (let employee of employees) {
+                    await OrganizationEmployee.softDelete(context, employee.id, DV_SENDER)
+                }
+
```
Comment (AntonAL) [inline apps/condo/domains/organization/schema/ResetOrganizationService.js:104]: You also need to delete `OrganizationEmployeeRole`

Comment (VKislov) [reply apps/condo/domains/organization/schema/ResetOrganizationService.js:104]: @pahaz explained the essence of this mutation as deleting an organization with all integrations disabled, i.e. deleting their connection contexts. Removal of such OrganozationEmployeeRole and specializations will be handled by a different logic. Later, when the cascading deletion of related entities appears, when deleting an employee, these entities will be cleaned up

File: apps/condo/domains/organization/schema/ResetOrganizationService.js (hunk 5 above)
Comment (akarjakin) [reply apps/condo/domains/organization/schema/ResetOrganizationService.js:24]: This is constant, it should be ERRORS, codegen behaviour will change soon. Can you please fix? :)

File: apps/condo/domains/organization/schema/ResetOrganizationService.js [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+                const organizationEntity = await getById('Organization', organizationId)
+                if (!organizationEntity) {
+                    throw new GQLError(errors.ORGANIZATION_NOT_FOUND, context)
+                }
+
+                const properties = await loadListByChunks({
+                    context,
+                    list: Property,
+                    chunkSize: 20,
+                    limit: 100000,
```
Comment (akarjakin) [inline apps/condo/domains/organization/schema/ResetOrganizationService.js:81]: looks like lots of very potentially heavy operations. can we move all these logic to async task ?

Comment (akarjakin) [reply apps/condo/domains/organization/schema/ResetOrganizationService.js:81]: We use to move heavy calculations/requests to async tasks in order not to block API for too long.

**Pull request merged**